### PR TITLE
8308808: SunMSCAPI public keys returns internal key array

### PR DESCRIPTION
**Pull request by blperez01**
Changed `getEncoded` for both EC and RSA to return a copy of the internal key array to avoid mutability issues.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8308808](https://bugs.openjdk.org/browse/JDK-8308808): SunMSCAPI public keys returns internal key array (**Bug** - P4)

### Reviewers
 * [Sean Mullan](https://openjdk.org/census#mullan) (@seanjmullan - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/14425/head:pull/14425` \
`$ git checkout pull/14425`

Update a local copy of the PR: \
`$ git checkout pull/14425` \
`$ git pull https://git.openjdk.org/jdk.git pull/14425/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 14425`

View PR using the GUI difftool: \
`$ git pr show -t 14425`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/14425.diff">https://git.openjdk.org/jdk/pull/14425.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/14425#issuecomment-1587911562)

### CODE REVIEW

File: src/jdk.crypto.mscapi/windows/classes/sun/security/mscapi/CPublicKey.java [hunk 1]
```diff
@@ -93,7 +93,10 @@ public byte[] getEncoded() {
                     // ignore
                 }
             }
-            return encoding;
+
+            byte[] safeEncoding = new byte[encoding.length];
+            System.arraycopy(encoding, 0, safeEncoding, 0, encoding.length);
+            return safeEncoding;
```
Comment (seanjmullan) [inline src/jdk.crypto.mscapi/windows/classes/sun/security/mscapi/CPublicKey.java:99]: You can replace these lines (and below) with `return encoding.clone();` which is more compact and the same AFAICT.

Comment (blperez01) [reply src/jdk.crypto.mscapi/windows/classes/sun/security/mscapi/CPublicKey.java]: Good call, just pushed those changes.

File: test/jdk/sun/security/mscapi/EncodingMutability.java [hunk 2]
```diff
@@ -0,0 +1,25 @@
+/*
+ * @test
+ * @bug 8308808
+ * @requires os.family == "windows"
+ * @modules jdk.crypto.mscapi/sun.security.mscapi
```
Comment (seanjmullan) [inline test/jdk/sun/security/mscapi/EncodingMutability.java:5]: I think you just need `jdk.crypto.mscapi` on this line as you are not accessing the internal `sun.security.mscapi` classes.

File: test/jdk/sun/security/mscapi/EncodingMutability.java
Comment (seanjmullan) [inline test/jdk/sun/security/mscapi/EncodingMutability.java:1]: Missing Oracle/OpenJDK copyright. Just copy from another test and change date to 2023.

File: test/jdk/sun/security/mscapi/EncodingMutability.java [hunk 3]
```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright (c) 2005, 2023, Oracle and/or its affiliates. All rights reserved.
```
Comment (seanjmullan) [inline test/jdk/sun/security/mscapi/EncodingMutability.java:2]: Should only have the initial year the code was created and the last year that the code is updated, so in this case it should just have 2023.

**Pull request closed without merging**